code/code/misc/player_data.cc

```diff
@@ -1084,7 +1084,7 @@ void do_the_player_stuff(const char *name)
     return;
 
   // skip faction data
-  if (strlen(name) > 6 && !strcmp(&name[strlen(name) - 8], ".faction"))
+  if (strlen(name) > 8 && !strcmp(&name[strlen(name) - 8], ".faction"))
```

Comment:
why not sstring(name).endsWith(".faction")? if it doesn't exist, let's write it, instead of perpetuating this eyebleed.

Reply:
> why not sstring(name).endsWith(".faction")?

Entirely because I haven't learned how sstrings work yet and didn't really understand what this code is doing. 

I just saw what appeared to be an easy fix due to a typo that was causing some runtime errors and was attempting to stabilize more than optimize.

Reply:
I'm worried here. We're making a change without understanding the intent of the original code - and we assume it works purely because the compiler doesn't warn anymore.

`strlen()` is rarely a good thing to call - code should be high-level, read like a book. `foo.endsWith(".faction") reads easily. Even if your language (plain old C, suppose) doesn't support strings / methods, you should still have a function  `string_endsWith(foo, ".faction")` or similar.

Reply:
I just had a look at the code - turns out that this `do_the_player_stuff` (genius name, huh?) is called for every file in player files. Therefore `name` is a filename, and the only sensible thing you can do with a filename and `.faction` is to check whether the filename ends with `.faction`. Therefore, even without hunting down the bytes and offsets and pointers in that (frankly foul-smelling) `strlen()`, we can assume that it's checking for the ending.

We have a string library in Sneezy, in `code/sys/sstring.h` and `code/sys/sstring.cc`, so what I'd do in this situation is slap on another method into .h, implement it in the cc, and write a unittest in `code/tests/SString.h`, then replace that strlen-monstrosity with a nice and readable boolean check.